qt/python/mantidqt/mantidqt/widgets/codeeditor/multifileinterpreter.py

```diff
@@ -66,6 +66,12 @@ def __init__(self, font=None, default_content=None, parent=None, completion_enab
         # setting defaults
         self.confirm_on_save = True
 
+        # Object to monitor file changes
+        self.file_watcher = QFileSystemWatcher()
+        self.file_watcher.fileChanged.connect(self.file_changed_event)
+        self.files_changed_unhandled = set()
+        self.monitor_files_changing = True
```

Comment:
I think a comment above these two variables would help to make what's happening clearer

Reply:
I'll add a comment

Reply:
```suggestion
        # When monitor_files_changing is True, then any files that are detected
        # as modified by the QFileSystemWatcher will be added to the set. We 
        # need to disable the file monitoring when we want to save the file so 
        # we don't get notified about our own save.
        self.files_changed_unhandled = set()
        self.monitor_files_changing = True
```